### PR TITLE
V3 pending

### PR DESCRIPTION
**Pull request by zbraniecki**

### CODE REVIEW

File: src/bindings/html/dom.js [hunk 1]
```diff
@@ -82,6 +88,11 @@ function camelCaseToDashed(string) {
 }
 
 export function translateElement(element) {
+  if (!this.service.languages) {
```
Comment (zbraniecki) [inline src/bindings/html/dom.js]: agh, and the problem I hit here is that in non-pretranslated case, translateDocument is fired before service.languages is set :/

**Comment by stasm** [comment]
Hmm, at the risk of sounding ignorant:  what is the purpose of `pendingElements` in v3 api which is always async?  I get that in v2 we needed to track sync `translateElement` calls made before the context was ready.  I think we might not need that at all in v3.  Perhaps one thing we could change is to call `view.observe()` on view initialization.  We disconnect the observer anyways when the translation is applied, so there's no risk of catching our own mutations.

**Comment by zbraniecki** [comment]
that's a good point. The only question is, what happens when we fire formatValue (in translateElement) when service.languages is undefined? Can we somehow make it a promise that will resolve later?

**Comment by stasm** [comment]
You shouldn't be able to do that because `this.languages` is created in https://github.com/l20n/l20n.js/blob/75768ae42a1bfcfd70c40394d8f231b8d0d870bb/src/runtime/web/index.js#L47, a mere two instructions after `document.l10n` is created.

**Comment by stasm** [comment]
And both of those things are created in `init` which is run when `readyState` changes to `interactive`.  Any other code inserting elements into DOM should run after this, too, and after l20n.js.

**Comment by stasm** [comment]
So, to recap:

    <script defer src="l20n.js"></script>
    <script defer src="inserThingstIntoDOM.js"></script>

We should document this as *the* way to plug l20n.js and other scripts into HTML.

**Comment by zbraniecki** [comment]
you are right. We just need to start observing in View.init

**Comment by stasm** [comment]
Great, that's less code to write! :)  Do you want to make this change in this PR?

**Comment by zbraniecki** [comment]
yeah, I'll do this when we land the pretranslate

**Comment by stasm** [comment]
Could we land this in parallel?  Up to you.

**Pull request closed without merging**